TESTING.md

```diff
@@ -140,4 +140,24 @@ mypy .
 
 from the root. See `mypy.ini` for our custom mypy settings.
 
+## Generate documentation from Jupyter notebooks
 
+The documents in the `tutorials` section are automatically generated from the Jupyter notebooks.
+
+The `convert.sh` script in the `scripts/nb2doc/` directory can be used:
+
+* to generate documents from new notebooks;
+* to ensure that any changes to the existing notebooks are reflected in the existing documents (for instance in a CI setting).
+
+The script must be run from the project root directory and requires [Pandoc](https://pandoc.org/) to be already installed.
+
+```bash
+$ scripts/nb2doc/convert.sh
```

Comment:
I get errors for `Unknown option --markdown-headings.
Try pandoc --help for more information.`.

pandoc version 2.9.2.1 
What are you using? Maybe we should specify which version we expect this to work with

Reply:
Ok, installing the latest version `3.0.1` works and produces the same output.

However there is some warnings - [NbConvertApp] Converting notebook examples/node-regression-with-subgraph-and-graph-sample.ipynb to asciidoc
``` MissingIDFieldWarning: Code cell is missing an id field, this will become a hard error in future nbformat versions. You may want to use `normalize()` on your notebooks before validations (available since nbformat 5.1.4). Previous versions of nbformat are fixing this issue transparently, and will stop doing so in the future.
  validate(nb)```

Reply:
I am using version 2.19.2 so I was thinking to add a note with `Pandoc >= 2.19`, but if you've tried Pandoc 3 I can take a look and make it conformant (so we can say `Pandoc >= 3.0`).

Reply:
Actually it seems that Pandoc 3+ is not yet supported by `nbconvert`:

`Your version must be at least (1.12.1) but less than (3.0.0).`

Running it with 3.0.1 anyway still works and does not produce any changes to the generated docs. We can still stick to the latest pre-3.x version (2.19.2) for the time being.

Reply:
Indeed. For our purposes, when running in CI we'd still be able to use the Pandoc version we need and, in case, update it later.